standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -983,12 +983,11 @@ public void alter_catalog(AlterCatalogRequest rqst) throws TException {
     GetCatalogResponse oldCat = null;
 
     try {
+      ms.openTransaction();
```

Comment:
why do we need to open txn before the `PreAlterCatalogEvent`? are those transactional as well?